src/library_postprocessing.rs

```diff
@@ -168,4 +170,107 @@ impl Library {
             _ => false,
         }
     }
+
+    fn resolve_class_structs(&mut self) {
+        // stores pairs of (gtype-struct-c-name, type-name)
+        let mut structs_and_types = Vec::new();
+
+        for (ns_id, ns) in self.namespaces.iter().enumerate() {
+            for type_ in ns.types.iter() {
+                let type_ = type_.as_ref().unwrap(); //Always contains something
+
+                match *type_ {
+                    Type::Record(ref record) => {
+                        if let Some(ref struct_for) = record.gtype_struct_for {
+                            if let Some(struct_for_tid) = self.find_type(ns_id as u16, struct_for) {
+                                structs_and_types.push ((record.c_type.clone(), struct_for_tid));
+                            }
+                        }
+                    },
+
+                    _ => (),
+                }
+            }
+        }
+
+        for (gtype_struct_c_type, struct_for_tid) in structs_and_types {
+            match *self.type_mut(struct_for_tid) {
+                Type::Class(ref mut klass) => {
+                    klass.c_class_type = Some(gtype_struct_c_type);
+                },
+
+                Type::Interface(ref mut iface) => {
+                    iface.c_class_type = Some(gtype_struct_c_type);
+                },
+
+                ref x @ _ => unreachable!("Something other than a class or interface has a class struct: {:?}", x)
+            }
+        }
+    }
+
+    fn correlate_class_structs(&self) {
+        for (ns_id, ns) in self.namespaces.iter().enumerate() {
+            for type_ in ns.types.iter() {
+                let type_ = type_.as_ref().unwrap(); //Always contains something
+                let mut name;
+                let mut type_struct;
+                let mut c_class_type;
+
+                match *type_ {
+                    Type::Class(ref klass) => {
+                        name         = &klass.name;
+                        type_struct  = &klass.type_struct;
+                        c_class_type = &klass.c_class_type;
+                    },
+
+                    Type::Interface(ref iface) => {
+                        name         = &iface.name;
+                        type_struct  = &iface.type_struct;
+                        c_class_type = &iface.c_class_type;
+                    },
+
+                    _ => { continue; }
+                }
+
+                if let &Some(ref type_struct) = type_struct {
```

Comment:
Please `if let Some(ref type_struct) = *type_struct {` if compiler allow it.

Reply:
Done.  Good catch.